backend/tracim_backend/lib/translate/services/systran.py

```diff
@@ -67,8 +70,26 @@ def _translate_file(
         )
         if response.status_code == HTTPStatus.OK:
             return response.raw
+        elif response.status_code == HTTPStatus.FORBIDDEN:
+            raise TranslationServiceAccessRefused("access refused to systran translation service")
         else:
-            raise TranslationFailed(str(response.json()))
+            try:
+                error = response.json()["error"]
+            except Exception as exc:
+                raise TranslationServiceException(
+                    "Unknown error when requesting translation server"
+                ) from exc
+
+            if error.get("statusCode") == HTTPStatus.BAD_REQUEST:
+                raise InvalidParametersForTranslationService(
+                    "Invalid parameters given for translation: {}".format(error["message"])
+                )
+            elif error.get("statusCode") == HTTPStatus.INTERNAL_SERVER_ERROR:
+                raise TranslationServiceServerError(
+                    "Translation service server error: {}".format(error["message"])
+                )
```

Comment:
Note: The 500 case (properly handled with json body) is far from exceptional to get in systran api. If you try to translate something systran is not able to translate (because of missing combination, like same to same language, or just because it don't get the data to do so: you can translate "caribou" in japanese for example)